web/src/components/upgrade_service/ConfirmAndDeploy.tsx

```diff
@@ -0,0 +1,353 @@
+import { useEffect, useState } from "react";
+import Modal from "react-modal";
+// TODO: add type checking support for react-remarkable or add a global ignore
+// @ts-ignore
+import Markdown from "react-remarkable";
+import { useLocation, useNavigate, useParams } from "react-router-dom";
+
+import { KotsPageTitle } from "@components/Head";
+import Icon from "@components/Icon";
+import SkipPreflightsModal from "@components/shared/modals/SkipPreflightsModal";
+import PreflightsProgress from "@components/troubleshoot/PreflightsProgress";
+import { useApps } from "@features/App";
+import { useDeployAppVersion } from "./hooks/index";
+import {
+  useGetPrelightResults,
+  // useIgnorePermissionErrors,
+} from "@features/PreflightChecks/api";
+import { KotsParams } from "@types";
+
+import "../../scss/components/PreflightCheckPage.scss";
+
+interface PreflightResultResponse {
+  learnMoreUri: string;
+  message: string;
+  title: string;
+  showCannotFail: boolean;
+  showFail: boolean;
+  showPass: boolean;
+  showWarn: boolean;
+}
+
+const ConfirmAndDeploy = ({
+  setCurrentStep,
+}: {
+  setCurrentStep: (step: number) => void;
+}) => {
+  useEffect(() => {
+    setCurrentStep(2);
+  }, []);
+  const navigate = useNavigate();
+  const [
+    showContinueWithFailedPreflightsModal,
+    setShowContinueWithFailedPreflightsModal,
+  ] = useState(false);
+  const [
+    showConfirmIgnorePreflightsModal,
+    setShowConfirmIgnorePreflightsModal,
+  ] = useState(false);
+
+  const { sequence = "0", slug } = useParams<keyof KotsParams>() as KotsParams;
+  const { mutate: deployKotsDownstream } = useDeployAppVersion({
+    slug,
+    sequence,
+  });
+  // TODO: figure out what this is for
```

Comment:
this is for minimal RBAC, which isn't supported in embedded cluster. we can deal with it in the future when we apply this workflow for existing cluster installations.